staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor_test.go

```diff
@@ -0,0 +1,788 @@
+	panic("unimplemented")
+}
+
+func (p anyObject) MarshalCBOR() ([]byte, error) {
+	return encMode.Marshal(p.Value)
+}
+
+func (p *anyObject) UnmarshalCBOR(in []byte) error {
+	return decMode.Unmarshal(in, &p.Value)
+}
+
+func TestEncode(t *testing.T) {
+	for _, tc := range []struct {
+		in       interface{}
+		expected string
+	}{
+		{
+			in:       int64(0),
+			expected: "00",
+		},
+		{
+			in:       int64(1),
+			expected: "01",
+		},
+		{
+			in:       int64(10),
+			expected: "0a",
+		},
+		{
+			in:       int64(23),
+			expected: "17",
+		},
+		{
+			in:       int64(24),
+			expected: "1818",
+		},
+		{
+			in:       int64(25),
+			expected: "1819",
+		},
+		{
+			in:       int64(100),
+			expected: "1864",
+		},
+		{
+			in:       int64(1000),
+			expected: "1903e8",
+		},
+		{
+			in:       int64(1000000),
+			expected: "1a000f4240",
+		},
+		{
+			in:       int64(1000000000000),
+			expected: "1b000000e8d4a51000",
+		},
+		{
+			in:       int64(-1),
+			expected: "20",
+		},
+		{
+			in:       int64(-10),
+			expected: "29",
+		},
+		{
+			in:       int64(-100),
+			expected: "3863",
+		},
+		{
+			in:       int64(-1000),
+			expected: "3903e7",
+		},
+		{
+			in:       0.0,
+			expected: "f90000",
+		},
+		{
+			in:       math.Copysign(0, -1),
+			expected: "f98000",
+		},
+		{
+			in:       1.0,
+			expected: "f93c00",
+		},
+		{
+			in:       1.1,
+			expected: "fb3ff199999999999a",
+		},
+		{
+			in:       1.5,
+			expected: "f93e00",
+		},
+		{
+			in:       65504.0,
+			expected: "f97bff",
+		},
+		{
+			in:       100000.0,
+			expected: "fa47c35000",
+		},
+		{
+			in:       3.4028234663852886e+38,
+			expected: "fa7f7fffff",
+		},
+		{
+			in:       1.0e+300,
+			expected: "fb7e37e43c8800759c",
+		},
+		{
+			in:       5.960464477539063e-8,
+			expected: "f90001",
+		},
+		{
+			in:       0.00006103515625,
+			expected: "f90400",
+		},
+		{
+			in:       -4.0,
+			expected: "f9c400",
+		},
+		{
+			in:       -4.1,
+			expected: "fbc010666666666666",
+		},
+		// TODO: Should Inf/-Inf/NaN be supported? Current Protobuf will encode this, but
+		// JSON will produce an error.  This is less than ideal -- we can't transcode
+		// everything to JSON.
+		{
+			in:       math.Inf(1),
+			expected: "f97c00",
+		},
+		{
+			in:       math.NaN(),
+			expected: "f97e00",
+		},
+		{
+			in:       math.Inf(-1),
+			expected: "f9fc00",
+		},
+		{
+			in:       false,
+			expected: "f4",
+		},
+		{
+			in:       true,
+			expected: "f5",
+		},
+		{
+			in:       nil,
+			expected: "f6",
+		},
+		{
+			in:       []interface{}{},
+			expected: "80",
+		},
+		{
+			in:       []interface{}{1, 2, 3},
+			expected: "83010203",
+		},
+		{
+			in:       []interface{}{1, []interface{}{2, 3}, []interface{}{4, 5}},
+			expected: "8301820203820405",
+		},
+		{
+			in:       []interface{}{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25},
+			expected: "98190102030405060708090a0b0c0d0e0f101112131415161718181819",
```

Comment:
I'm feeling slow trying to understand this. This is the stringification of the encoded bytes of a slice of integers?

Reply:
I'm going to separate the tests that come directly from Appendix A of the CBOR RFC and clarify them. This one makes more sense in that context. It's an example of an array whose length is too long to be encoded in the "additional information" bits of the initial byte, and short enough to be encoded in a single byte.